x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedConfigTests.java

```diff
@@ -167,6 +180,33 @@ public void testFutureMetadataParse() throws IOException {
         assertNotNull(DatafeedConfig.LENIENT_PARSER.apply(parser, null).build());
     }
 
+    public void testToXContentForClusterState() throws IOException {
```

Comment:
I wanted to add a test to check the datafeed type isn't written when we are writing to cluster state i.e. this piece of code in DatafeedConfig::toXContent
```
if (params.paramAsBoolean(ToXContentParams.FOR_CLUSTER_STATE, false) == false) {
            builder.field(CONFIG_TYPE.getPreferredName(), TYPE);
        }
``` 

But it's hard to test so I added a check for the headers which similarly should only be persisted in certain circumstances

Reply:
Why not the following something similar to the following? It may work to simple directly check the serialized data string.

```
String stringToCheck = "\""+DatafeedConfig.CONFIG_TYPE.getPreferredName()+"\":\""+DatafeedConfig.TYPE+"\"";
assertThat(forClusterstateXContent.utf8ToString(), containsString(stringToCheck));
...
assertThat(forClusterstateXContent.utf8ToString(), not(containsString(stringToCheck)));
```

Reply:
Yes that works, great idea I never thought of it.